### PR TITLE
Fibonacci series implementation without using recursion in C++

### PR DESCRIPTION
**Pull request by imsushant12**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

I have added the Fibonacci series implementation without using recursion.

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#5718

### CODE REVIEW

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

Use proper display message how you are reading and writing outputs

File: C-Plus-Plus/math/fibonacci_space_optimized.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#include <bits/stdc++.h>
+using namespace std;
+
+/*
+ Fibonacci(n) = Fibonacci(n-1) + Fibonacci(n-2). Rather than
+ string each value, we will only store the last and second
+ last number of the Fibonacci series.
+ Initially we will store first and second value i.e. 1
+ and run the loop from third place till the number given.
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/fibonacci_space_optimized.cpp]: Shift this above the header

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
What exactly you are doing finding fibonacci sum or what? didn't get exact clarity you have written something and finding something else

File: C-Plus-Plus/math/fibonacci_space_optimized.cpp (hunk 1 above)
Comment (imsushant12) [reply C-Plus-Plus/math/fibonacci_space_optimized.cpp]: Shifted

**Comment by imsushant12** [comment]
> What exactly you are doing finding fibonacci sum or what? didn't get exact clarity you have written something and finding something else

I am finding fibonacci number at given position (for example n=5  then answer will be fib(5) = 5; So I am using the DP approach i.e. saving the last and second last number to get current number.

File: C-Plus-Plus/math/fibonacci_space_optimized.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    Time Complexity : O(n)
+    Space Complexity : O(1)
+
+    Sample Outputs:
+    
+    Test Case - 1:
+    Enter number : 5
+    Fibonacci : 5
+
+    Test Case - 1:
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/fibonacci_space_optimized.cpp]: test case 2 not 1

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rename your issue, PR and file name it not suits with the work you have done

File: C-Plus-Plus/math/fibonacci_space_optimized.cpp (hunk 2 above)
Comment (imsushant12) [reply C-Plus-Plus/math/fibonacci_space_optimized.cpp]: done!

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**